plugins/inputs/nsdp/nsdp.go

```diff
@@ -0,0 +1,121 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package nsdp
+
+import (
+	_ "embed"
+	"net"
+	"strconv"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/tdrn-org/go-nsdp"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+const pluginName = "nsdp"
```

Comment:
Please directly use the string at the correct places instead of defining a constant.

Reply:
Adapted via 8c321f127f426a5c1cbdc9876b56cf4409fafce2